pkg/workload/workload.go

```diff
@@ -39,6 +40,40 @@ var (
 	admissionManagedConditions = []string{kueue.WorkloadAdmitted, kueue.WorkloadEvicted}
 )
 
+type FlavorResourceQuantities map[kueue.ResourceFlavorReference]map[corev1.ResourceName]int64
+type LastScheduleClusterQueueState struct {
+	LastScheduledFlavorIdx map[string]map[corev1.ResourceName]int
+	ResourceFlavors        map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor
+	ClusterQueueUsage      FlavorResourceQuantities
+	CohortUsage            FlavorResourceQuantities
+}
+
+func (s *LastScheduleClusterQueueState) Equal(resourceFlavors map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor, clusterQueueUsage, cohortUsage FlavorResourceQuantities) bool {
```

Comment:
If going with this full snapshot approach, can't we do more with that data, so the assignment from the beginning is triggered only the current state of the queue/cohort gives better chances to fit in a "better" flavor.

Reply:
I will record the generation of the Cohort and cluster queue, and increase the generation when more resources were add to cohort or cluster queue (either some workloads were released or cluster queue/cohort was changed)